lib/src/supabase_auth.dart

```diff
@@ -151,6 +203,88 @@ class SupabaseAuth {
       return _authCallbackUrlHostname == uri.host;
     }
   }
+
+  /// Enable deep link observer to handle deep links
+  void _startDeeplinkObserver() {
+    Supabase.instance.log('***** SupabaseDeepLinkingMixin startAuthObserver');
+    _handleIncomingLinks();
+    _handleInitialUri();
+  }
+
+  /// Stop deep link observer
+  ///
+  /// Automatically called on dispose().
+  void _stopDeeplinkObserver() {
+    Supabase.instance.log('***** SupabaseDeepLinkingMixin stopAuthObserver');
+    if (_deeplinkSubscription != null) _deeplinkSubscription?.cancel();
+  }
+
+  /// Handle incoming links - the ones that the app will recieve from the OS
+  /// while already started.
+  void _handleIncomingLinks() {
+    if (!kIsWeb) {
+      // It will handle app links while the app is already started - be it in
+      // the foreground or in the background.
+      _deeplinkSubscription = uriLinkStream.listen(
+        (Uri? uri) {
+          if (uri != null) {
+            _handleDeeplink(uri);
+          }
+        },
+        onError: (Object err) {
+          _onErrorReceivingDeeplink(err.toString());
+        },
+      );
+    }
+  }
+
+  /// Handle the initial Uri - the one the app was started with
+  ///
+  /// **ATTENTION**: `getInitialLink`/`getInitialUri` should be handled
+  /// ONLY ONCE in your app's lifetime, since it is not meant to change
+  /// throughout your app's life.
+  ///
+  /// We handle all exceptions, since it is called from initState.
+  Future<void> _handleInitialUri() async {
+    if (!shouldHandleInitialDeeplink) return;
```

Comment:
`shouldHandleInitialDeeplink` is a getter that checks to see the value of another variable, `_initialDeeplinkIsHandled`.

I think we can probably just use 1 variable for this.
Like, we can just check `_initialDeeplinkIsHandled` or set `shouldHandleInitialDeeplink` to true/false.

Is there a specific reason we wanted to separate them?

Reply:
@DanMossa Great point! Fixing this now!